### PR TITLE
Fix a crash when H3 transaction ends early

### PR DESCRIPTION
**Pull request by maskit**
Similar to #9715 but this one is for H3 layer.

### CODE REVIEW

File: proxy/http3/Http3Transaction.cc [hunk 1]
```diff
@@ -342,30 +429,16 @@ Http3Transaction::~Http3Transaction()
 }
 
 int
-Http3Transaction::state_stream_open(int event, void *edata)
+Http3Transaction::state_stream_open(int event, Event *edata)
 {
   // TODO: should check recursive call?
-  if (this->_thread != this_ethread()) {
```
Comment (maskit) [inline proxy/http3/Http3Transaction.cc:348]: Note for reviewers: This block was copied from H2 but thread mismatch should not happen. If it happens, we should fix the code that schedules events on a wrong thread.

File: proxy/http3/Http3Transaction.cc [hunk 2]
```diff
@@ -234,24 +229,112 @@ HQTransaction::direction() const
...
+    return;
+  }
+
+  this->_write_complete_event = this->_thread->schedule_imm(this, VC_EVENT_WRITE_COMPLETE, &this->_write_vio);
+}
+
+void
+HQTransaction::_unschedule_write_complete_event()
+{
+  if (this->_read_complete_event) {
```
Comment (brbzull0) [inline proxy/http3/Http3Transaction.cc:327]: shouldn't we be checking `_write_complete_event`?

Comment (maskit) [reply proxy/http3/Http3Transaction.cc]: Yes. Thanks.

**Review by brbzull0 — APPROVED** [review]
looks good to me. Thanks.

**Pull request merged**